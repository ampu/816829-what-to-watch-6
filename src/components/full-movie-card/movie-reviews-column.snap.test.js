import React from 'react';
import range from 'lodash.range';

import {render} from '@testing-library/react';

import MovieReviewsColumn from './movie-reviews-column';

const REVIEWS_COUNT = 3;
const RANGE_OFFSET = 1;

describe(`MovieReviewsColumn component`, () => {
  it(`Should render according to the snapshot with no reviews provided`, () => {
    const {container} = render(<MovieReviewsColumn/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot with reviews provided`, () => {
    const reviews = range(RANGE_OFFSET, RANGE_OFFSET + REVIEWS_COUNT).map((id) => ({
      id: id.toString(),
      user: {
        id: `fakeUserId`,
        name: `fakeUserName`,
      },
      rating: 9.9,
      text: `fakeText`,
      date: `2020-03-09`,
    }));
    const {container} = render(<MovieReviewsColumn reviews={reviews}/>);
    expect(container).toMatchSnapshot();
  });
});
