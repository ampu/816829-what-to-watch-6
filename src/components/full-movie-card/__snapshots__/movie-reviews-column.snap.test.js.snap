// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieReviewsColumn component Should render according to the snapshot with no reviews provided 1`] = `
<div>
  <div
    class="movie-card__reviews-col"
  />
</div>
`;

exports[`MovieReviewsColumn component Should render according to the snapshot with reviews provided 1`] = `
<div>
  <div
    class="movie-card__reviews-col"
  >
    <div
      class="review"
    >
      <blockquote
        class="review__quote"
      >
        <p
          class="review__text"
        >
          fakeText
        </p>
        <footer
          class="review__details"
        >
          <cite
            class="review__author"
          >
            fakeUserName
          </cite>
          <time
            class="review__date"
            datetime="2020-03-09"
          >
            March 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        class="review__rating"
      >
        9,9
      </div>
    </div>
    <div
      class="review"
    >
      <blockquote
        class="review__quote"
      >
        <p
          class="review__text"
        >
          fakeText
        </p>
        <footer
          class="review__details"
        >
          <cite
            class="review__author"
          >
            fakeUserName
          </cite>
          <time
            class="review__date"
            datetime="2020-03-09"
          >
            March 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        class="review__rating"
      >
        9,9
      </div>
    </div>
    <div
      class="review"
    >
      <blockquote
        class="review__quote"
      >
        <p
          class="review__text"
        >
          fakeText
        </p>
        <footer
          class="review__details"
        >
          <cite
            class="review__author"
          >
            fakeUserName
          </cite>
          <time
            class="review__date"
            datetime="2020-03-09"
          >
            March 09, 2020
          </time>
        </footer>
      </blockquote>
      <div
        class="review__rating"
      >
        9,9
      </div>
    </div>
  </div>
</div>
`;
